client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -108,6 +136,72 @@ const EventForm: React.FC<EventFormProps> = (props) => {
           </FormControl>
         )}
 
+        <FormControl id="first-name" isRequired>
+          <Box display="flex" alignItems="end" m="1">
+            <FormLabel> Sponsers</FormLabel>
+            <Spacer />
+            <Button
+              onClick={() => {
+                append({
+                  type: 'FOOD',
+                  id: 0,
```

Comment:
```suggestion
                  type: firstSponsorType,
                  id: firstSponsorId,
```
This way we know both `type` and `id` are valid in isolation.